src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -1643,29 +1754,581 @@ protected Object parseValue(CommandSource source, CommandArgs args) throws Argum
         @Override
         protected Object getValue(String choice) throws IllegalArgumentException {
             UUID uuid = UUID.fromString(choice);
+            boolean found = false;
             for (World world : Sponge.getServer().getWorlds()) {
                 Optional<Entity> ret = world.getEntity(uuid);
                 if (ret.isPresent()) {
-                    return ret.get();
+                    Entity entity = ret.get();
+                    if (this.checkEntity(entity)) {
+                        return entity;
+                    }
+                    found = true;
                 }
             }
+            if (found) {
+                throw new IllegalArgumentException("Input value " + choice + " was not an entity of the required type!");
```

Comment:
It's theoretically possible by random chance, much more likely when mapmakers use specific ones on purpose. For instance, imagine a server responsible for managing several instances of a parkour map, which used UUID'd armor stands for fancy effects.